### PR TITLE
[DI] Fix false-positive circular exception

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #25240
| License       | MIT
| Doc PR        | -

### CODE REVIEW

**Comment by ElectricMaxxx** [comment]
With this my 3.4 build would have been red too, right? Or can i reset my "fix" (set the second service to public) now?

**Comment by nicolas-grekas** [comment]
yes, you can revert your fix

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Comment by ElectricMaxxx** [comment]
Thanks @nicolas-grekas

File: src/Symfony/Component/DependencyInjection/Tests/Fixtures/php/services_almost_circular_public.php [hunk 1]
```diff
@@ -93,6 +95,26 @@ protected function getBar3Service()
...
+    /**
+     * Gets the public 'bar5' shared service.
+     *
+     * @return \stdClass
+     */
+    protected function getBar5Service()
+    {
+        $a = ${($_ = isset($this->services['foo5']) ? $this->services['foo5'] : $this->getFoo5Service()) && false ?: '_'};
+
+        if (isset($this->services['bar5'])) {
```
Comment (stof) [inline src/Symfony/Component/DependencyInjection/Tests/Fixtures/php/services_almost_circular_public.php]: shouldn't this come first, before defining the `$a` variable ?

Comment (stof) [reply src/Symfony/Component/DependencyInjection/Tests/Fixtures/php/services_almost_circular_public.php]: ah no, understand. This is the call to ` $this->getFoo5Service()` which can register it in `$this->services`

**Pull request merged**